packages/components/src/DataList/components/DataListLayoutInternal/DataListHeader.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { DataListLayoutInternal } from "./DataListLayoutInternal";
+import { Breakpoints } from "../../DataList.const";
+import styles from "../../DataList.css";
+import {
+  DataListHeader as DataListHeaderType,
+  DataListItemTypeFromHeader,
+  DataListObject,
+  DataListProps,
+} from "../../DataList.types";
+import { sortSizeProp } from "../../DataList.utils";
+import { DataListLayoutProps } from "../DataListLayout/DataListLayout";
+
+export function DataListHeader<T extends DataListObject>({
+  layouts,
+  mediaMatches,
+  headerData,
+  headerVisibility,
+}: {
+  layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined;
+  mediaMatches?: Record<Breakpoints, boolean>;
+  headerData?: DataListItemTypeFromHeader<T, DataListHeaderType<T>>;
+  headerVisibility: NonNullable<DataListProps<T>["headerVisibility"]>;
+}) {
```

Comment:
nit: mostly because we've been consisitent around putting these things on interfaces.

protip: highlight +  `cmd + .` > extract

<img width="335" alt="image" src="https://github.com/GetJobber/atlantis/assets/15986172/ad181df7-f4de-4aeb-9e30-1e14460c3b4f">